### PR TITLE
Update build pipeline clean workspace to delete hidden files

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Previous clean workspace PR updated to use rm -rf *, which does not include hidden files.
This PR includes a post clean with cleanWs to clear the hidden files.

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
run tests

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1017,13 +1026,29 @@ class Build {
...
+                                    // Clean remaining hidden files using cleanWs
+                                    try {
+                                        context.println "Cleaning workspace hidden files using cleanWs: " + context.WORKSPACE
+                                        context.cleanWs notFailBuild: true, disableDeferredWipeout: true, deleteDirs: true
+                                    } catch (e) {
+                                        context.println "Failed to clean ${e}"
+                                    }
                                 } else if (cleanWorkspaceBuildOutputAfter) {
-                                    context.println "Cleaning workspace build output files: " + context.WORKSPACE + "/workspace/build/src/build"
+                                    context.println "Cleaning workspace build output non-hidden files: " + context.WORKSPACE + "/workspace/build/src/build"
```
Comment (sxa) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1040]: Why are these necessary? We're not using `/*` on these so the `rm -rf ` shoulkd be sufficient

**Review by sxa — CHANGES_REQUESTED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1040]: so you mean, just use the above rm -rf ?
since there should not be any hidden files in the build output?
so remove the cleanWs for this bit?

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:1040]: oh ! Doh! I see what you mean :-) (slow today!)

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: removed

**Comment by andrew-m-leonard** [comment]
run tests

**Review by M-Davies — APPROVED** [review]

**Review by gdams — APPROVED** [review]
